test/CMakeLists.txt

```diff
@@ -22,12 +22,22 @@ configure_lit_site_cfg(
 set(NVQPP_TEST_PARAMS
   nvqpp_site_config=${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg.py)
 
+# Gather all the runtime libraries to make sure they are up-to-date.
+# FIXME: These libraries should self-register to a CMake property so here we
+# can just use a macro instead of listing them.
+set(TEST_CUDAQ_LIBRARIES
+  cudaq cudaq-common cudaq-mlir-runtime cudaq-builder cudaq-ensmallen
+  cudaq-nlopt cudaq-spin
+  cudaq-em-default cudaq-em-photonics cudaq-platform-default
+  nvqir nvqir-qpp nvqir-dm)
```

Comment:
Should we consider the creation a cmake function/macro, say `cudaq_library`, for creating libraries ? This would prevent the mistake of creating new runtime libs and forgetting to add them to the here.

Reply:
Yes, this is exactly what is meant by the fixme above. I'll file an issue.

Reply:
My apologies, I didn't register the FIXME message after reading it